### DISCUSSION TITLE
scratchclient help

### DISCUSSION THREAD

**Discussion by CubeyTheCube** [CubeyTheCube/scratchclient#21] [General]
ask stuff about scratchclient here

**Comment by catkitty12021** [comment]
How do you post a comment in a studio?

**Reply by CubeyTheCube** [reply]
Right now, you can use 
```python
session.get_studio(id).post_comment("comment")
```
keep in mind that it uses the scratchr2 studio endpoint so it can be subject to bugs

**Comment by ShardOSDev** [comment]
How would i do error handling when someone puts in invalid credentials.

**Reply by Quantum-Codes** [reply]
Like what? Username/password? Scratchclient should throw an error if it's invalid

**Reply by ShardOSDev** [reply]
I figured it out.

**Reply by Quantum-Codes** [reply]
Ok

**Comment by EntentaPL** [comment]
I have this:

![image](https://user-images.githubusercontent.com/84442980/146594754-3dfa19e7-fbf9-4a4e-ae47-c513930341ed.png)

In screenshot is a variable 'data' with a number, but when I want to run script I have a error, that cloud can only send a numbers. What I can to do to python interpret the variable value, **not name**.

Script with importing scratch client and login to scratch is made.

**Reply by Quantum-Codes** [reply]
Scratch takes numbers in str format. Can u send the code maybe?

**Reply by EntentaPL** [reply]
Okay, but I censor password.

```python
from scratchclient import ScratchSession
session = ScratchSession("Unixy-Code" , "password")
connection = session.create_cloud_connection(617406446)


import geocoder
g = geocoder.ip('me')
data= (g.latlng)


connection.set_cloud_variable("cloud", data)
```

Please, don't ask to me why I do this. It's only test to do other thing.

**Reply by Quantum-Codes** [reply]
Okay. I see that data is a list. So you can use an encoder(like you do in scratch)

**Reply by Quantum-Codes** [reply]
```python

from scratchclient import ScratchSession
session = ScratchSession("Unixy-Code" , "password")
connection = session.create_cloud_connection(617406446)


import geocoder
g = geocoder.ip('me')
data= (g.latlng)

from ScraGet import ScraGet
encoder = ScraGet.encoder()
x = encoder.encode(data) #encoding the list

connection.set_cloud_variable("cloud", x)
```
This should work. Read this page to know how to use this and decode on scratch too https://github.com/Quantum-Codes/ScraGet/wiki/Cloud-Encoder

**Reply by EntentaPL** [reply]
Thanks, I'll check.

**Comment by piw-piw** [comment]
How to use UserProfile module?

**Comment by pikachub2005** [comment]
Just here to see if  anyone remembers me ;-;

**Reply by Quantum-Codes** [reply]
I do!

**Comment by Mr-Sk1ttle** [comment]
how do you see who commented last on a project?

**Reply by CubeyTheCube** [reply]
Use
 ```python
session.get_project(123456789).get_comments(limit=1)[0].username
```

**Comment by Kupicx** [comment]
How do I get comments from a studio?

**Reply by CubeyTheCube** [reply]
Unfortunately, this is not possible at the moment. I have an update to scratchclient that's almost done, that includes the ability to get studio and profile comments.
